ios/brave-ios/Sources/AIChat/Components/Feedback/AIChatFeedbackView.swift

```diff
@@ -0,0 +1,381 @@
+              }
+            }
+          })
+        
+        if text.isEmpty {
+          VStack {
+            Text(Strings.AIChat.feedbackInputViewTitle)
+              .font(.subheadline)
+              .foregroundColor(Color(braveSystemName: .textTertiary))
+              .disabled(true)
+              .allowsHitTesting(false)
+              .padding(.vertical, 8.0)
+              .padding(.horizontal, 5.0)
+            Spacer()
+          }
+        }
+      }
+      .fixedSize(horizontal: false, vertical: true)
+      .padding(.trailing)
+      
+      Button {
+        Task { @MainActor in
+          let permissionStatus = await model.speechRecognizer.askForUserPermission()
+          if permissionStatus {
+            model.isVoiceEntryPresented = true
+            model.activeInputView = .feedbackView
+          } else {
+            model.isNoMicrophonePermissionPresented = true
+            model.activeInputView = .none
+          }
+        }
+      } label: {
+        Image(braveSystemName: "leo.microphone")
+          .foregroundStyle(Color(braveSystemName: .iconDefault))
+      }
+      .hidden(isHidden: !model.speechRecognizer.isVoiceSearchAvailable)
+    }
+    .padding(12.0)
+    .overlay {
+      RoundedRectangle(cornerRadius: 8.0, style: .continuous)
+        .stroke(Color(braveSystemName: .iconDefault), lineWidth: 0.5)
+    }
+    .background(Color(braveSystemName: .containerBackground))
+    .clipShape(RoundedRectangle(cornerRadius: 8.0, style: .continuous))
+    .onReceive(model.speechRecognizer.$finalizedRecognition) { recognition in
+      if recognition.status && model.activeInputView == .feedbackView {
+        // Feedback indicating recognition is finalized
+        AudioServicesPlayAlertSound(SystemSoundID(kSystemSoundID_Vibrate))
+        UIImpactFeedbackGenerator(style: .medium).bzzt()
+        
+        // Update Prompt
+        text = recognition.searchQuery
+        
+        // Clear the SpeechRecognizer
+        model.speechRecognizer.clearSearch()
+        model.isVoiceEntryPresented = false
+        model.activeInputView = .none
+      }
+    }
+  }
+}
+
+private struct AIChatFeedbackLeoPremiumAdView: View {
+  let openURL: (URL) -> Void
+
+  var body: some View {
+    Text(Strings.AIChat.feedbackPremiumAdTitle)
+      .tint(Color(braveSystemName: .textInteractive))
+      .font(.subheadline)
+      .foregroundStyle(Color(braveSystemName: .textSecondary))
+      .fixedSize(horizontal: false, vertical: true)
+      .frame(maxWidth: .infinity, alignment: .leading)
+      .padding()
+      .background(
+        LinearGradient(gradient:
+                        Gradient(colors: [
+                          Color(UIColor(rgb: 0xF8BEDA)).opacity(0.1),
+                          Color(UIColor(rgb: 0xAD99FF)).opacity(0.1)]),
+                       startPoint: .init(x: 1.0, y: 1.0),
+                       endPoint: .zero)
+      )
+      .clipShape(RoundedRectangle(cornerRadius: 8.0, style: .continuous))
```

Comment:
`.background(LinearGradient(...), in: RoundedRectangle(cornerRadius: 8.0, style: .continuous))`